src/DCDataSet.cpp

```diff
@@ -36,13 +38,52 @@
 
 namespace splash
 {
+    herr_t H5Ewalk_strings(unsigned n, const H5E_error2_t *err_desc, void *client_data)
+    {
+        std::ostringstream *stream = (std::ostringstream *)client_data;
+        const char		*maj_str = NULL;
+        const char		*min_str = NULL;
+
+        /* Check arguments */
+        assert (err_desc);
+        if (!client_data) client_data = stderr;
+
+        /* Get descriptions for the major and minor error numbers */
+        maj_str = H5Eget_major (err_desc->maj_num);
+        min_str = H5Eget_minor (err_desc->min_num);
```

Comment:
These strings must be freed with `H5free_memory` (see https://www.hdfgroup.org/HDF5/doc/Advanced/FreeingMemory/FreeingMemoryAllocatedByTheHdf5Library.pdf)